web/src/TargetIpsPopup.jsx

```diff
@@ -20,62 +20,80 @@
  */
 
 import React, { useEffect, useState } from "react";
-import { useInstallerClient } from "./context/installer";
-import Popup from "./Popup";
 import { Button, List, ListItem, Text } from "@patternfly/react-core";
+import Popup from "./Popup";
 
-const initIpData = {
-  addresses: [],
-  hostname: ""
-};
-
-function formatIp(address, prefix) {
-  return address + "/" + prefix;
-}
+import { useInstallerClient } from "./context/installer";
+import { useCancellablePromise } from "./utils";
+import { formatIp } from "./client/network";
 
 export default function TargetIpsPopup() {
   const client = useInstallerClient();
-  const [state, setState] = useState(initIpData);
+  const { cancellablePromise } = useCancellablePromise();
+  const [connections, setConnections] = useState([]);
+  const [hostname, setHostname] = useState("");
   const [isOpen, setIsOpen] = useState(false);
-  const { hostname, addresses } = state;
 
   useEffect(() => {
-    client.network.config().then((data) => {
-      setState(data);
+    cancellablePromise(client.network.config()).then(config => {
+      setConnections(config.connections);
+      setHostname(config.hostname);
     });
+  }, [client.network, cancellablePromise]);
+
+  useEffect(() => {
+    const onConnectionAdded = addedConnection => {
+      setConnections(conns => [...conns, addedConnection]);
+    };
+
+    return client.network.listen("connectionAdded", onConnectionAdded);
+  }, [client.network]);
+
+  useEffect(() => {
+    const onConnectionRemoved = connectionPath => {
+      setConnections(conns => conns.filter(c => c.path !== connectionPath));
+    };
+
+    return client.network.listen("connectionRemoved", onConnectionRemoved);
+  }, [client.network]);
+
+  useEffect(() => {
+    const onConnectionUpdated = updatedConnection => {
+      setConnections(conns => {
+        const newConnections = conns.filter(c => c.path !== updatedConnection.path);
+        return [...newConnections, updatedConnection];
+      });
+    };
+
+    return client.network.listen("connectionUpdated", onConnectionUpdated);
   }, [client.network]);
 
-  const ips = addresses.map((addr) => formatIp(addr.address, addr.prefix));
-  let label = ips[0];
-  let title = "IP addresses";
+  if (connections.length === 0) return null;
+
+  const ips = connections.flatMap(conn => conn.addresses.map(formatIp));
+  const [firstIp] = ips;
 
-  if (hostname) {
-    label += ` (${hostname})`;
-    title += ` for ${hostname}`;
-  }
+  if (ips.length === 0) return null;
 
   const open = () => setIsOpen(true);
   const close = () => setIsOpen(false);
 
-  if (addresses.length === 0) return null;
-
-  if (ips.length === 1) return <Text component="small" className="host-ip">{label}</Text>;
-
   return (
     <>
-      <Button variant="link" onClick={open}>
-        { label }
+      <Button variant="link" onClick={open} isDisabled={ips.length === 1}>
+        {firstIp} {hostname && <Text component="small">({hostname})</Text>}
```

Comment:
:warning:  **NOTE** 

Despite we have a _Network overview_ in the installation summary now, we have kept this component because it is interesting to have a way to quickly identify the machine we have dealing with. For example, think of someone performing multiple installations at the same time… they must be able to distinguish between these machines no matter which section is displayed (product selection, storage section, etc)

Of course, we can do a lot of things to improve the current presentation, but let's book some time to think/discuss about it later.